### PR TITLE
autopep8: avoid running on non-committed files.

### PR DESCRIPTION
**Pull request by bneradt**
It was found that if someone has non-commited Python files, such as can
happen if they have a virtual environment in their source tree, autopep8
will inspect those as well. This is slow and probably not desired by the
user.

This patch only runs autopep8 on files tracked by git. It will also
enable autopep8 to check .test.ext extensions, thus the updates to those
files.

It also has autopep8 run silently so it doesn't produce as much noise.

Considering the above, the only "interesting" change here to review is the 
change to tools/autopep8.sh.

### CODE REVIEW

File: tools/autopep8.sh [hunk 1]
```diff
@@ -59,6 +59,32 @@ function main() {
   fi
 
   DIR=${@:-.}
+
+  # Only run autopep8 on tracked files. This saves time and possibly avoids
+  # formatting files the user doesn't want formatted.
+  tmp_dir=/tmp/git_files_$$
```
Comment (randall) [inline tools/autopep8.sh:65]: this should probably use `mktemp -d` (eg `tmp_dir=$(mktemp -d)`)

Comment (bneradt) [reply tools/autopep8.sh]: Yeah, that's better.

**Review by randall — APPROVED** [review]

**Review by SolidWallOfCode — APPROVED** [review]

**Comment by bneradt** [comment]
I'll create a separate PR for 9.0.x.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**